XUtils.h

```diff
@@ -86,4 +86,13 @@ static inline unsigned int prevPowerOf2(unsigned int x) {
 unsigned int prevPowerOf2(unsigned int x);
 #endif
 
+static inline int popCount8(unsigned char x) {
+#ifdef __POPCNT__ // Defined if x86 POPCNT instruction is enabled ("-mpopcnt")
```

Comment:
I don't quite like relying on compiler behaviour to define this constant instead of checking the availability of the proper built-in in `configure.ac`. Such a check could even handle a tristate variable like:
0. Use custom shift&multiply implementation
1. Use builtin when `popcnt` instruction available
2. Use builtin always when `__builtin_popcount` is supported

Reply:
@BenBE I can't check the `popcnt` instruction in configure time. That the compiler supports `popcnt` instruction doesn't mean it would be enabled in CFLAGS. It has to be the build time check.

Reply:
Just a random try, but seems to work as expected …
```diff
$ git diff
diff --git a/configure.ac b/configure.ac
index c356fb40..497a421c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -201,6 +201,19 @@ AC_COMPILE_IFELSE([
    AC_MSG_RESULT(yes)],
    AC_MSG_RESULT(no))
 
+AC_MSG_CHECKING(for popcount instruction)
+AC_COMPILE_IFELSE([
+   AC_LANG_SOURCE(
+      [
+         #ifndef __POPCNT__
+         #error POPCNT instruction not supported ...
+         #endif
+      ],[]
+   )],
+   [AC_DEFINE([HAVE_INSTR_POPCOUNT], 1, [Define to 1 if the compiler advertises `POPCNT' instruction to be available.])
+   AC_MSG_RESULT(yes)],
+   AC_MSG_RESULT(no))
+
 # ----------------------------------------------------------------------
 
 
```

Implementing a more complex option for the user to specifx which code variant he likes/prefers, is not too complicated either.

Reply:
@BenBE You are just converting a macro constant to another name. For the sake of checking it in `configure`. When I said it has to be a build time check, I mean the `CFLAGS` variable can be overridden at `make` time, making your configure-time variable `HAVE_INSTR_POPCOUNT` inaccurate in representing what processor instructions are available.